proto/src/determined/trial/v1/trial.proto

```diff
@@ -87,6 +87,29 @@ message Trial {
   CheckpointWorkload best_checkpoint = 10;
 }
 
+// TrialProfilerMetricSummary is a summary of a single metric.
+message TrialProfilerMetricSummary {
```

Comment:
I don't think this needs to be so generic. IMO the response JSON is really convoluted here and the list of summaries are returned by the API are undocumented. I feel like the API would be more usable if we didn't use this generic approach.

The current response as a JSON is

```json
{
    "summary": {
        "metric_summaries": [
            {
                "average": 1234,
                "stable": true,
                "labels": {
                    "trial_id": 1,
                    "name": "samples_per_second",
                    "agent_id": null,
                    "gpu_uuid": null,
                    "metric_type": "PROFILER_METRIC_TYPE_MISC",
                }
            }
        ]
    }
}
```

The same information represented more succinctly:

```json
{
    "average_throughput": 1234,
    "throughput_is_stable": true
}
```

Or if you want a hierarchy

```json
{
    "throughput": {
        "average": 1234,
        "stable": true
    }
}
```

You shouldn't have to iterate through a list to parse the response from this API - the set of fields we return is static, so we should use a well-defined response schema.

Reply:
Considering there has been some interest in generic summarizations already, I'd rather keep this in place. now adding GPU util/mem summaries in the backend is a ~3 line code change.

Reply:
What about keeping the generic summarization object, but using a map instead of a list? e.g.

```json
{
  "throughout": {
        ... The summary object
    }
}
```
Designing this API to return a fixed size list instead of a map doesn't make a ton of sense to me.

Reply:
sure i can take that

Reply:
well, i will but unfortunately because of grpc's rigidity this means we lose the generic nature internally a bit without some not-worth-it hacks.

Reply:
@armandmcqueen after some thought over the weekend, I'm beginning to think trying to show "stable" is just a mistake.. what about just error bars?